DscResources/MSFT_GroupResource/MSFT_GroupResource.psm1

```diff
@@ -2511,7 +2515,18 @@ function Add-GroupMember
         $MemberAsPrincipal
     )
 
-    $Group.Members.Add($MemberAsPrincipal)
+    try
+    {
+        $Group.Members.Add($MemberAsPrincipal)
+    }
+    catch
+    {
+        Write-Verbose -Message $script:localizedData.PrincipalCollectionAddMethodException
```

Comment:
Would this be a condition that should be displayed as a Warning or is this something we're going to consider as a "normal" condition that simply occurs?

Reply:
You mean to use Write-Warning cmdlet instead of Write-Verbose? Yeah, I guess it fits better. Updated it.

Reply:
Although, on a second thought. My whole intention of outputting those messages was to let user know that execution flow went this route, that exception has occurred, but we were able to recover from it by switching to another method. From this point of view it doesn't really make any difference if we report it as a warning or as a normal message. Whether this issue occurs or not can't really be decided by a user. It's not like a user is going to make his groups contain only members from own AD forest just to suppress the warnings. Therefore the purpose of the messages is to show the execution flow for those who are interested, and for that Write-Verbose fits better.